### PR TITLE
LPS-129712 upload guest file on form widget

### PR DESCRIPTION
**Pull request by linolaoj**

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -243,7 +243,9 @@ protected String getGuestUploadURL(
 			RequestBackedPortletURLFactoryUtil.create(httpServletRequest);
 
 		PortletURL portletURL = requestBackedPortletURLFactory.createActionURL(
-			DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM);
+			GetterUtil.getString(
+				portal.getPortletId(httpServletRequest),
+				DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM));
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java:248]: @linolaoj, why this change is necessary? Can you please clarify?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 2]
```diff
@@ -252,10 +252,10 @@ protected String getGuestUploadURL(
...
 		portletURL.setParameter(
 			"formInstanceId",
-			ParamUtil.getString(httpServletRequest, "formInstanceId",
-				Long.toString(
-					GetterUtil.getLong(
-						ddmFormFieldRenderingContext.getProperty("formInstanceId")))));
+			ParamUtil.getString(
+				httpServletRequest, "formInstanceId",
+				String.valueOf(
+					ddmFormFieldRenderingContext.getDDMFormInstanceId())));
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java:258]: Nice improvement, @linolaoj! :clap:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java [hunk 3]
```diff
@@ -223,8 +223,8 @@ public String getContainerId() {
 		DDMFormRenderingContext ddmFormRenderingContext =
 			createDDMFormRenderingContext(ddmForm);
 
-		ddmFormRenderingContext.addProperty("formInstanceId",
-			ddmFormInstance.getFormInstanceId());
+		ddmFormRenderingContext.addProperty(
+			"formInstanceId", ddmFormInstance.getFormInstanceId());
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java:227]: @linolaoj, shouldn't you use the method `setDDMFormInstanceId` that you created on this commit?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/DDMFormRenderingContext.java [hunk 4]
```diff
@@ -48,6 +48,10 @@ public String getContainerId() {
 		return _containerId;
 	}
 
+	public long getDDMFormInstanceId() {
+		return _ddmFormInstanceId;
+	}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/DDMFormRenderingContext.java:53]: @linolaoj, where this method is being used on this pull request? Can you please clarify?

**Comment by natocesarrego** [comment]
@linolaoj, I'm closing this pull due to the conflicting files, please resend.

Thanks.

**Comment by linolaoj** [comment]
resending here https://github.com/liferay-forms/liferay-portal/pull/184
